### ISSUE TITLE
Exception during compilation: (ArgumentError) argument error calling :erlang.get_module_info(Plug.Conn, :deprecated)

### ISSUE THREAD

**Issue by kipcole9** [elixir-lang/elixir#7182]
### Environment

Elixir 1.6.0-rc.0

### Current behavior

Compiling a dependency (ex_cldr which I maintain) produced this error:
```
$ mix deps.clean ex_cldr
* Cleaning ex_cldr
$ mix deps.get
...
$ mix deps.compile ex_cldr
==> ex_cldr
Compiling 2 files (.erl)
Compiling 30 files (.ex)
Generating Cldr for 2 locales named ["en-001", "root"] with a default locale named "en-001"
could not compile dependency :ex_cldr, "mix compile" failed. You can recompile this dependency with "mix deps.compile ex_cldr", update it with "mix deps.update ex_cldr" or clean it with "mix deps.clean ex_cldr"
** (ArgumentError) argument error
    :erlang.get_module_info(Plug.Conn, :deprecated)
    Plug.Conn.__info__/1
    (mix) lib/mix/tasks/xref.ex:335: Mix.Tasks.Xref.load_exports_and_deprecated/1
    (mix) lib/mix/tasks/xref.ex:326: Mix.Tasks.Xref.load_exports_and_deprecated_into_acc/2
    (elixir) lib/enum.ex:1899: Enum."-reduce/3-lists^foldl/2-0-"/3
    (mix) lib/mix/tasks/xref.ex:295: anonymous fn/2 in Mix.Tasks.Xref.source_warnings/2
    (elixir) lib/enum.ex:1899: Enum."-reduce/3-lists^foldl/2-0-"/3
    (mix) lib/mix/tasks/xref.ex:291: Mix.Tasks.Xref.source_warnings/2
```
After conversation with @michalmuskala is seems this may have been the result of compilation mismatch between elixir versions and the user who reported the error is not long able to reproduce.

Looking at the master branch in `elixir_erl` that generate the `__info__` function I can't see how this error would be produced which is another reason why i think it might be a version mismatch.

Nevertheless, it would seem reasonable to avoid calling ` :erlang.get_module_info(SomeModule, :deprecated)` with keys that are not recognised by the version.  If it is version mismatch then I guess this would be a bug fix to the 1.5 branch rather than 1.6

### Expected behavior

Don't execute ` :erlang.get_module_info(ModName, key)` when the key is unknown in the current Elixir version.

**Comment by josevalim** [comment]
We already have everything in place for this not to happen. Mix
automatically deletes all previous .beam files when the Elixir version
changes.

If there is a bug in our safety mechanisms, we would need a mechanism to
reproduce in order to fix it.

I guess for now we can change the v1.6 branch exclusively to have an extra
safety net, but it will be removed on v1.7.
-- 


*José Valimwww.plataformatec.com.br
<http://www.plataformatec.com.br/>Founder and Director of R&D*

**Comment by michalmuskala** [comment]
I guess one way this could happen is with some module coming from outside of a mix project - through archive or even something else. I know that wasn't the case in this situation, though.

**Comment by josevalim** [comment]
Fixed on v1.6.
